[BUG] Fix `MultiRocketMultivariate` failures

### Describe the bug

`test_multirocket_multivariate_on_basic_motions` occasionally fails due to a difference in accuracy despite a `random_state` being set.

`MultiRocketHydraRegressor` fails `test_fit_deterministic` with multivariate data, the `HydraRegressor` does not, so I assume this is the multirocket side.

### Steps/Code to reproduce the bug

Run `MultiRocketHydraRegressor` `fit` and `predict` twice and check results.

### Expected results

Tests pass.

### Actual results

Results are different causing test failures.

### Versions

_No response_

I can confirm the bug is reproducible locally using :

```python
from aeon.transformations.collection.convolution_based import (
     MultiRocketMultivariate
)
from aeon.classification.convolution_based import RocketClassifier
from aeon.datasets import load_basic_motions
from numpy.testing import assert_array_almost_equal, assert_equal

for i in range(1000):
    X_train, y_train = load_basic_motions(split="train")
    X_test, y_test = load_basic_motions(split="test")

    trf = MultiRocketMultivariate(num_kernels=512, random_state=10)
    trf.fit(X_train)
    X_train_T = trf.transform(X_train)
    X_test_T = trf.transform(X_test)

    trf2 = MultiRocketMultivariate(num_kernels=512, random_state=10)
    trf2.fit(X_train)
    X_train_T2 = trf.transform(X_train)
    X_test_T2 = trf.transform(X_test)

    assert_array_almost_equal(X_train_T2, X_train_T)
    assert_array_almost_equal(X_test_T2, X_test_T)
    
    rkt1 = RocketClassifier(
        rocket_transform="multirocket", num_kernels=100, random_state=10
    )
    rkt1.fit(X_train, y_train)
    acc1 = rkt1.score(X_test, y_test)
    
    rkt2 = RocketClassifier(
        rocket_transform="multirocket", num_kernels=100, random_state=10
    )
    rkt2.fit(X_train, y_train)
    acc2 = rkt2.score(X_test, y_test)
    
    assert_equal(acc1, acc2)
``` 
Results in :
```python-traceback
AssertionError: 
Items are not equal:
 ACTUAL: 1.0
 DESIRED: 0.975
```

The two object rkt1/rkt2 have different kernel parameter despite random_state value. This is indeed because we do not set random state inside python function `get_parameter` before calling a randomly setting `num_channels_per_combination`.
Setting a random state locally fixed the issue. 

We should rather initialize a numpy Random Generator in init and use this generator for all random usage.

--- a/aeon/transformations/collection/convolution_based/_multirocket_multivariate.py
+++ b/aeon/transformations/collection/convolution_based/_multirocket_multivariate.py
@@ -112,6 +112,9 @@ class MultiRocketMultivariate(BaseCollectionTransformer):
         -------
         self
         """
+        if self.random_state is not None:
+            np.random.seed(self.random_state)
+
         if self.normalise:
             X = (X - X.mean(axis=-1, keepdims=True)) / (
                 X.std(axis=-1, keepdims=True) + 1e-8

